commands/cli.go

```diff
@@ -0,0 +1,290 @@
+			},
+			Action: func(c *cli.Context) error {
+
+				err := validateContext(c, "send")
+				if err != nil {
+					log.Fatalf("%s", err)
+				}
+				var plaintext []byte
+
+				// attempt stdin scan, SEND should be pipeable for things like cat'ing a file.
+				plaintext = stdinScan()
+				if plaintext == nil {
+					plaintext = utils.UserInput()
+				}
+				if len(plaintext) == 0 {
+					log.Fatal("Empty input detected. Aborting")
+				}
+
+				client := cloud.NewGCPKMS(c.String("key"))
+				cypher, err := client.Encrypt(plaintext)
+				if err != nil {
+					log.Fatal(err)
+				}
+				encoded := base64.StdEncoding.EncodeToString(cypher)
+
+				fmt.Println("Hello, I've shared some data with you with sctl! https://github.com/vapor-ware/sctl")
+				fmt.Println("Once installed, run the following commands to view this sensitive information")
+				fmt.Println("")
+				fmt.Println("```")
+				cmd := fmt.Sprintf("sctl receive --key=%s %s", c.String("key"), encoded)
+				fmt.Println(cmd)
+				fmt.Println("```")
+				return nil
+			},
+		},
+		{
+			Name:  "receive",
+			Usage: "Read a plaintext encoded secret",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:   "key",
+					EnvVar: "SCTL_KEY",
+					Usage:  "Gcloud KMS Key URI",
+				},
+			},
+			Action: func(c *cli.Context) error {
+				if len(c.Args()) >= 1 {
+					decoded, err := base64.StdEncoding.DecodeString(c.Args().First())
+					if err != nil {
+						log.Fatal(err)
+					}
+					client := cloud.NewGCPKMS(c.String("key"))
+					cypher, err := client.Decrypt(decoded)
+					if err != nil {
+						log.Fatal(err)
+					}
+
+					fmt.Println(string(cypher))
+				}
+				return nil
+			},
+		},
+		{
+			Name:  "rm",
+			Usage: "rm a secret",
+			Action: func(c *cli.Context) error {
+				secretName := strings.ToUpper(c.Args().First())
+				utils.DeleteSecret(secretName)
+				return nil
+			},
+		},
+		{
+			Name:  "list",
+			Usage: "list known secrets",
+			Action: func(c *cli.Context) error {
+				secrets := utils.ReadSecrets()
+				var knownKeys []string
+				for _, secret := range secrets {
+					knownKeys = append(knownKeys, secret.Name)
+				}
+				sort.Strings(knownKeys)
+				for _, k := range knownKeys {
+					fmt.Println(k)
+				}
+				return nil
+			},
+		},
+		{
+			Name:           "run",
+			Usage:          "run a command with secrets exported as env",
+			SkipArgReorder: true,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:   "key",
+					EnvVar: "SCTL_KEY",
+					Usage:  "GCloud KMS Key URI",
+				},
+			},
+			Action: func(c *cli.Context) error {
+				validateContext(c, "run")
+
+				var secrets []utils.Secret
+				var arguments []string = c.Args()
+
+				cmd := exec.Command(arguments[0], arguments[1:]...)
+				cmd.Env = os.Environ()
+				secrets = utils.ReadSecrets()
+				for _, secret := range secrets {
+					// uncan the base64
+					decoded, err := base64.StdEncoding.DecodeString(secret.Cyphertext)
+					if err != nil {
+						log.Fatalf("CLI - DECODING - %s", err)
```

Comment:
I feel like this error message could be cleaned up / made more clear

Reply:
I actgually put these in here as such so i would have an idea where the errors were coming from since this is kind of a god module and its gross to debug.